conjur-api/Role.cs

```diff
@@ -0,0 +1,80 @@
+// <copyright file="Role.cs" company="Conjur Inc.">
+//     Copyright (c) 2018 Cyberark Ltd. All rights reserved.
+// </copyright>
+// <summary>
+//     Role manipulation routines.
+// </summary>
+
+namespace Conjur
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+
+    /// <summary>
+    /// Conjur Role reference.
+    /// </summary>
+    /// A role is an actor in the system, in the classical sense of role-based access control. Roles are the entities which receive permission grants.
+    public class Role : Resource
+    {
+        private readonly string path;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Conjur.Role"/> class.
+        /// </summary>
+        /// <param name="client">Conjur client to use to connect.</param>
+        /// <param name="kind">Kind of the role, for example 'group' or 'layer'.</param>
+        /// <param name="name">The Role name.</param>
+        internal Role(Client client, string kind, string name)
+            : base(client, kind, name)
+        {
+            this.path = $"authz/{client.GetAccountName()}/roles/{WebUtility.UrlEncode(kind)}/{WebUtility.UrlEncode(name)}";
+        }
+
+        /// <summary>
+        /// Check for the existence of a role.
+        /// </summary>
+        /// <returns>True if role exists otherwice false.</returns>
+        /// Only roles that you have read permission on will be searched.
```

Comment:
We want to leave standalone documentation lines? or add them to tags: summary / returns

Reply:
I'll wrap it with remarks tag (see https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/remarks)